test/clib/test_ctonedim.cpp

```diff
@@ -0,0 +1,305 @@
+#include <gtest/gtest.h>
+
+#include "cantera/core.h"
+#include "cantera/onedim.h"
+#include "cantera/thermo/ThermoFactory.h"
+
+extern "C" {
+    #include "cantera/clib/ct.h"
+    #include "cantera/clib/ctonedim.h"
+}
+
+using namespace Cantera;
+
+TEST(ctonedim, freeflow)
+{
+    soln_reset();
+
+    int sol = soln_newSolution("h2o2.yaml", "ohmech", "Mix");
+    int ph = soln_thermo(sol);
+    ASSERT_GE(ph, 0);
+    int kin = soln_kinetics(sol);
+    ASSERT_GE(kin, 0);
+    int tr = soln_transport(sol);
+    ASSERT_GE(tr, 0);
+
+    double T = 1050;
+    double P = 5 * 101325;
+    string X = "CH4:1.0, O2:2.0, N2:7.52";
+    thermo_setMoleFractionsByName(ph, X.c_str());
+    thermo_setTemperature(ph, T);
+    thermo_setPressure(ph, P);
+
+    int itype = 2; // free flow
+    int flow = stflow_new(ph, kin, tr, itype);
+    ASSERT_GE(flow, 0);
+
+    ASSERT_NEAR(stflow_pressure(flow), P, 1e-5);
+}
+
+TEST(ctonedim, inlet)
+{
+    int index = inlet_new();
+    ASSERT_GE(index, 0);
+}
+
+TEST(ctonedim, outlet)
+{
+    int index = outlet_new();
+    ASSERT_GE(index, 0);
+}
+
+TEST(ctonedim, reacting_surface)
+{
+    int index = reactingsurf_new();
+    ASSERT_GE(index, 0);
+
+    int gas = thermo_newFromFile("ptcombust.yaml", "gas");
+    int surf = thermo_newFromFile("ptcombust.yaml", "Pt_surf");
+    int kin = kin_newFromFile("ptcombust.yaml", "Pt_surf", surf, gas, -1, -1, -1);
+    ASSERT_GE(kin, 0);
+
+    int ret = reactingsurf_setkineticsmgr(index, kin);
+    ASSERT_EQ(ret, 0);
+}
+
+TEST(ctonedim, catcomb_stack)
+{
+    int sol = soln_newSolution("ptcombust.yaml", "gas", "");
+    int gas = soln_thermo(sol);
+    int gas_kin = soln_kinetics(sol);
+    int trans = soln_transport(sol);
+
+    int surf = thermo_newFromFile("ptcombust.yaml", "Pt_surf");
+    int surf_kin = kin_newFromFile("ptcombust.yaml", "Pt_surf", surf, gas, -1, -1, -1);
+
+    // inlet
+    int inlet = inlet_new();
+
+    // flow
+    int itype = 1; // free flow
+    int flow = stflow_new(gas, gas_kin, trans, itype);
+    domain_setID(flow, "flow");
+
+    // reacting surface
+    int reac_surf = reactingsurf_new();
+    int ret = reactingsurf_setkineticsmgr(reac_surf, surf_kin);
+    ASSERT_EQ(ret, 0);
+
+    // set up stack
+    std::vector<int> doms{inlet, flow, reac_surf};
+    int flame = sim1D_new(3, doms.data());
+    ASSERT_GE(flame, 0);
+    int dom = sim1D_domainIndex(flame, "flow");
+    ASSERT_GE(dom, 0);
+}
+
+vector<double> T_ctonedim = {
+    300.0000, 300.0000, 300.0000, 300.0001, 300.0010, 300.0141, 300.0842,
+    300.6108, 302.1595, 308.7286, 319.8533, 331.8250, 352.4362, 367.7950,
+    387.6687, 441.9631, 513.3673, 596.8101, 717.1222, 838.6055, 956.3989,
+    1067.1676, 1259.9339, 1408.2242, 1517.2894, 1598.5709, 1661.5194, 1712.0943,
+    1786.7072, 1843.4863, 1920.5714, 2009.4328, 2066.5316, 2135.8423, 2174.3517,
+    2207.6584, 2228.0248, 2228.0248};
```

Comment:
Fair point. The original intent was to ensure that both `cpp` and `clib` versions yield the same result. Whenever `SolutionArray` becomes available in `clib` (I have no plans to do this for 3.0), there's the option to have a reference solution instead. I guess it could be done with `sim1D_restore`, but I don't think it's necessary at this point.

Using the monotonic criterion with grid refinement disabled as suggested for the time being.